common/djangoapps/entitlements/utils.py

```diff
@@ -0,0 +1,47 @@
+from course_modes.models import CourseMode
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from student.models import CourseEnrollment
+
+
+def is_course_run_entitlement_fullfillable(course_run_id, compare_date, entitlement):
+    """
+    Checks that the current run meets the following criteria for an entitlement
+
+    1) Are currently running or start in the future
+    2) A user can enroll in
+    3) A user can upgrade to the entitlement mode
+    4) Are not already currently enrolled in a session
+
+    Arguments:
+        course_run_id (String): The id of the Course run that is being checked.
+        compare_date: The date and time that we are comparing against.  Generally the current date and time.
```

Comment:
Hm, I'm not a huge fan of adding a compare date here - are you doing it just because of tests? If so, I think altering the actual course info would be better for the tests. 

If you want to keep it this way, can you add to the comments: 'compare_date: The date and time that we are checking if the course entitlement is fulfillable. 

You can also rearrange the properties to -> (entitlement, course_run_id, compare_date = atetime.datetime.now(UTC)) so that you don't have to always pass it through.

Reply:
I added this because when it is called in the loop of course runsninbtbe original method I did not want to retrieve the current datetime  on every call of the method. I can add the default value though.